posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -16,6 +16,12 @@ import scala.scalanative.posix.sys.timeOps._
 import scala.scalanative.posix.netinet.{in, inOps}, in._, inOps._
 
 object SocketHelpers {
+  /*
+   * The following should be long enough and constant exists on macOS.
+   * https://www.gnu.org/software/libc/manual/html_node/Host-Identification.html
+   * https://man7.org/linux/man-pages/man2/gethostname.2.html
+   */
+  val MAXHOSTNAMELEN = 256.toUInt
```

Comment:
Excellent change!  This removes my concern about the largish stack allocation.

I noticed that ipToHost() in SocketHelpers.scala could benefit from
evolution & simplification.  If I ever get to a PR for that I might address
two pedantic points in this edit.  I am not asking for a change here, 
only capturing the info.

1) Would `final val MAXmumble` allow the compiler to generate better code
    (a constant? inlined?).  I have a vague memory of that.

2) Since this is a Posix helper, HOST_NAME_MAX in posix limits.h should
    be applicable.  However, that symbol does not exist in the SN implementation and would have to be added.  As said, subject for another PR.


1) Would `final val MAXmumble`  allow the SN compiler to make that a constant (and help inlining?).

2

Reply:
For future devos (tested on macOS):
```c
#include <unistd.h>
// host name len
printf("MAXHOSTNAMELEN: %d\n", MAXHOSTNAMELEN);
printf("HOST_NAME_MAX: %ld\n", sysconf(_SC_HOST_NAME_MAX));
```
Result:
```
MAXHOSTNAMELEN: 256
HOST_NAME_MAX: 255
```

Reply:
I really like the `sysconf()` approach.  It means a runtime call instead of a constant
but avoids a huge mess with pieces currently missing in SN.  Much easier to 
grok on-sight.

With some thought, he runtime call might be reduced to 1.

 Good information for a future PR.